### PR TITLE
DAC6-416 - Create page to display Taxpayers.

### PR DESCRIPTION
**Pull request by iservport**

### CODE REVIEW

File: app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    val controllerComponents: MessagesControllerComponents,
+    renderer: Renderer
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with NunjucksSupport {
+
+  private val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      // retrieve list of taxpayers TODO get the actual list from page
```
Comment (helensoniega) [inline app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala:57]: Is this comment still needed?

File: app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(SelectAnyTaxpayersThisEnterpriseIsAssociatedWithPage, value))
+            _              <- sessionRepository.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(SelectAnyTaxpayersThisEnterpriseIsAssociatedWithPage, mode, updatedAnswers))
+      )
+  }
+
+  def toCheckboxes(taxpayers: Set[JsObject]): Seq[Checkboxes.Checkbox] = taxpayers.map { json =>
+    val name = json.validate[Taxpayer].getOrElse(throw new IllegalArgumentException("")).nameAsString
+    Checkboxes.Checkbox(label = Literal(name), value = json.toString())
+  }.toSeq
```
Comment (helensoniega) [inline app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala:103]: Not used anywhere. It's similar to line 65 in the onPageLoad method above.

File: test/forms/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithFormProviderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  ".value" - {
+
+    val fieldName = "value"
+    val requiredKey = "selectAnyTaxpayersThisEnterpriseIsAssociatedWith.error.required"
+
+    "must bind correctly " in {
+
+      form.bind(Map("value[0]" -> "1")).get shouldEqual List("1")
+    }
+
```
Comment (helensoniega) [inline test/forms/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithFormProviderSpec.scala:35]: Test needed for mandatory page.

File: app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala (hunk 1 above)
Comment (iservport) [reply app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala:57]: Yes, it is. When we have the taxpayer sub-journey working, we will have actual data to display here.

File: app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala (hunk 2 above)
Comment (iservport) [reply app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala]: Removed

**Review by helensoniega — APPROVED** [review]

**Pull request merged**